meshmode/discretization/poly_element.py

```diff
@@ -530,16 +534,44 @@ def __init__(self, mesh_el_group, order, index=None, *, basis, unit_nodes):
             raise ValueError("basis dimension does not match element group: "
                     f"expected {mesh_el_group.dim}, got {basis._dim}.")
 
+        if isinstance(basis, mp.TensorProductBasis):
+            for b in basis.bases:
+                if b._dim != 1:
+                    raise ValueError("All bases used to construct the tensor "
+                                     "product must be of dimension 1. Support "
+                                     "for higher-dimensional component bases "
+                                     "does not yet exist.")
+
         if unit_nodes.shape[0] != mesh_el_group.dim:
             raise ValueError("unit node dimension does not match element group: "
-                    f"expected {mesh_el_group.dim}, got {unit_nodes.shape[0]}.")
+                    f"expected {self.mesh_el_group.dim}, "
+                    f"got {unit_nodes.shape[0]}.")
+
+        # NOTE there are cases where basis is a 1D `_SimplexONB` object. We wrap
+        # in a TensorProductBasis object if this is the case
+        if not isinstance(basis, mp.TensorProductBasis):
+            if basis._dim == 1 and unit_nodes.shape[0] == 1:
+                from modepy.modes import _get_orth_weight
+                basis = mp.TensorProductBasis([basis],
+                                                    _get_orth_weight([basis]))
```

Comment:
```suggestion
                basis = mp.TensorProductBasis([basis])
```
(see https://github.com/inducer/modepy/pull/72)